sdk-extensions/incubator/src/test/java/io/opentelemetry/sdk/extension/incubator/fileconfig/AttributeListFactoryTest.java

```diff
@@ -0,0 +1,138 @@
+                        new AttributeNameValueModel()
+                            .withName("doubleKey")
+                            .withValue(1.0d)
+                            .withType(AttributeNameValueModel.Type.DOUBLE),
+                        new AttributeNameValueModel()
+                            .withName("floatKey")
+                            .withValue(2.0f)
+                            .withType(AttributeNameValueModel.Type.DOUBLE),
+                        new AttributeNameValueModel()
+                            .withName("boolKey")
+                            .withValue(true)
+                            .withType(AttributeNameValueModel.Type.BOOL),
+                        new AttributeNameValueModel()
+                            .withName("strArrKey")
+                            .withValue(Arrays.asList("val1", "val2"))
+                            .withType(AttributeNameValueModel.Type.STRING_ARRAY),
+                        new AttributeNameValueModel()
+                            .withName("longArrKey")
+                            .withValue(Arrays.asList(1L, 2L))
+                            .withType(AttributeNameValueModel.Type.INT_ARRAY),
+                        new AttributeNameValueModel()
+                            .withName("intArrKey")
+                            .withValue(Arrays.asList(1, 2))
+                            .withType(AttributeNameValueModel.Type.INT_ARRAY),
+                        new AttributeNameValueModel()
+                            .withName("doubleArrKey")
+                            .withValue(Arrays.asList(1.0d, 2.0d))
+                            .withType(AttributeNameValueModel.Type.DOUBLE_ARRAY),
+                        new AttributeNameValueModel()
+                            .withName("floatArrKey")
+                            .withValue(Arrays.asList(1.0f, 2.0f))
+                            .withType(AttributeNameValueModel.Type.DOUBLE_ARRAY),
+                        new AttributeNameValueModel()
+                            .withName("boolArrKey")
+                            .withValue(Arrays.asList(true, false))
+                            .withType(AttributeNameValueModel.Type.BOOL_ARRAY)),
+                    mock(SpiHelper.class),
+                    Collections.emptyList()))
+        .isEqualTo(
+            io.opentelemetry.api.common.Attributes.builder()
+                .put("service.name", "my-service")
+                .put("strKey", "val")
+                .put("longKey", 1L)
+                .put("intKey", 2)
+                .put("doubleKey", 1.0d)
+                .put("floatKey", 2.0f)
+                .put("boolKey", true)
+                .put("strArrKey", "val1", "val2")
+                .put("longArrKey", 1L, 2L)
+                .put("intArrKey", 1, 2)
+                .put("doubleArrKey", 1.0d, 2.0d)
+                .put("floatArrKey", 1.0f, 2.0f)
+                .put("boolArrKey", true, false)
+                .build());
```

Comment:
I think it would read slightly cleaner to have these pulled up into a variable called `expected`.